pkg/api/types.go

```diff
@@ -551,6 +553,20 @@ type RBDVolumeSource struct {
 	ReadOnly bool `json:"readOnly,omitempty"`
 }
 
+// MetadataVolumeSource represents a volume containing metadata about a pod.
+type MetadataVolumeSource struct {
```

Comment:
It won't always be for metadata.  @sdminonne we should rename this `DownwardAPIVolumeSource`